tests/trickle/main.c

```diff
@@ -53,14 +53,16 @@ static void callback(void *args)
     return;
 }
 
-static trickle_t trickle = { .callback.func = &callback,
-                             .callback.args = NULL };
+static trickle_t trickle;
```

Comment:
What was the problem with this one ?

Reply:
the way of init the struct caused an error.

Reply:
maybe use

```C
static trickle_t trickle = { .callback = { .func = &callback,
                                           .args = NULL } };
```

instead? It doesn't change the behavior and is the more correct version.

Reply:
For native, iotlab-m3, samr21-xpro I get no error with the first version.

What was the error ? the commit message does not help me "tests/trickle: fix struct init"

Reply:
Error was reported by `clang` on macOS, which is often more pedantic than `gcc`

Reply:
mhm, can't reproduce right now - maybe it was reported by Murdock on some other platform (too). At least the old init version didn't work, @miri64 solution looks okay to me, too.